.github/workflows/integration.yml

```diff
@@ -161,63 +161,67 @@ jobs:
           echo "=============================== Node.js Vault CLI ============================"
           nodejs/dist/cli/vault.js --help
 
-      - name: Check Python version
-        run: vault --version
+        ######## CLI TESTS ########
 
       - name: Check stack status with Rust CLI
         run: bin/rust/vault stack
 
       - name: Store secret with Python
-        run: vault -s 'secret-python' -v 'sha-${{github.sha}}' -w
+        run: tests/store.sh vault secret-python "sha-${{github.sha}}"
 
       - name: Store secret with Go
-        run: bin/go/vault -s 'secret-go' -v 'sha-${{github.sha}}' -w
+        run: tests/store.sh bin/go/vault secret-go "sha-${{github.sha}}"
 
       - name: Store secret with Rust
-        run: bin/rust/vault -s 'secret-rust' -v 'sha-${{github.sha}}' -w
-
-      - name: Store secret with Nodejs
-        run: nodejs/dist/cli/vault.js s 'secret-nodejs' 'sha-${{github.sha}}' -w
+        run: tests/store.sh bin/rust/vault secret-rust "sha-${{github.sha}}"
 
-      - name: Validate storing worked Python
-        run: diff <(vault -l secret-python) <(echo -n sha-${{github.sha}})
+      - name: Store secret with Node.js
+        run: tests/store.sh nodejs/dist/cli/vault.js secret-nodejs "sha-${{github.sha}}"
 
-      - name: Validate Go and Rust secret equality with Python
-        run: diff <(vault -l secret-go) <(vault -l secret-rust)
+      - name: Lookup secret with Python
+        run: tests/lookup.sh vault secret-python "sha-${{github.sha}}"
 
-      - name: Validate Python secret equality with Rust and Go
-        run: diff <(bin/rust/vault -l secret-python) <(bin/go/vault -l secret-python)
+      - name: Lookup secret with Go
+        run: tests/lookup.sh bin/go/vault secret-go "sha-${{github.sha}}"
 
-      - name: Validate Rust and Python secret equality with Go and Nodejs
-        run: diff <(bin/go/vault -l secret-rust) <(nodejs/dist/cli/vault.js lookup secret-python)
+      - name: Lookup secret with Rust
+        run: tests/lookup.sh bin/rust/vault secret-rust "sha-${{github.sha}}"
 
-      - name: Validate Go and Python secret equality with Rust and Go
-        run: diff <(bin/rust/vault -l secret-go) <(bin/go/vault -l secret-python)
+      - name: Lookup secret with Node.js
+        run: tests/lookup.sh nodejs/dist/cli/vault.js secret-nodejs "sha-${{github.sha}}"
 
-      - name: Validate Python and Nodejs secret equality with Rust
-        run: diff <(bin/rust/vault -l secret-python) <(bin/rust/vault -l secret-nodejs)
+      - name: Validate storing worked
+        run: tests/validate.sh
```

Comment:
Could the `validate.sh` be given a list of "command to execute vault" as arguments and those would be tested? Here it's ambiguous which versions are being validated.

No need to do this here, just a thought :thought_balloon: